src/cascade/dismod/db/wrapper.py

```diff
@@ -209,13 +209,20 @@ def update_table_columns(self, table_name, table):
     def refresh(self, evict_tables=None):
         """ Throw away any un-flushed changes and reread data from disk.
         """
+        evicted = list()
         if evict_tables is None:
+            evicted = list(self._table_data.keys())
             self._table_data = {}
+            self._table_hash = {}
         else:
             to_evict = [evict_tables] if isinstance(evict_tables, str) else evict_tables
             for evict in to_evict:
                 if evict in self._table_data:
                     del self._table_data[evict]
+                if evict in self._table_hash:
+                    del self._table_hash[evict]
+                    evicted.append(evict)
```

Comment:
The previous fix I made neglected to also reset the table_hash. Maybe this is the final fix, but not sure.